Fixes network bugs and adjusts program access

NataKilar opened the pull request:
Fixes a few bugs with networks and their related machinery, and adds an option for grant based program access
* Fixes mainframes going invisible after opening their panel once
* Fixes Network Locks not having their UI properly updated.
* Changes how the transfer files backend works. Now it by default transfers the file with an option for cloning the file instead. Beforehand it used store_file to add the datum to the file list of whatever was being copied to, without removing itself from the former location, which resulted in every program downloaded from a mainframe being one instance. This lead to issues like #1037, which is now fixed.
* Fixes a bug where running a program without proper access would break the computer until it was restarted.
* Added an option for program access to be determined by grants through the access controller. If/when full replacement of the access system with grants occurs, this should probably be reworked., but it works for now.

afterthought2 commented on code/modules/modular_computers/file_system/program.dm:
```diff
@@ -83,32 +83,43 @@
+				accesses_to_check = access_controller.get_program_access(filename)
+				if(!length(accesses_to_check))
+					accesses_to_check = list(required_access)
+				else if(accesses_to_check[1] == "NONE")
```
Really not a fan of dummy NONE access like this. Empty list should signify this. If you need to inherit default access some but not all of the time, add a bool and/or proc to the program itself and check/call that to get extra default accesses with a different inheritance structure.

NataKilar commented:
Adjusted program access/grants to use the normal empty list paradigm. Also went back and adjusted programs to use lists for their required_access var, and cleaned up a couple oddities (Command and Communications and Deck Management) programs who used unique access checks that could easily be replaced.

afterthought2 commented on code/modules/modular_computers/file_system/programs/generic/deck_management.dm:
```diff
@@ -7,6 +7,7 @@
 	filename = "deckmngr"
 	filedesc = "Deck Management"
 	nanomodule_path = /datum/nano_module/deck_management
+	required_access = list(access_cargo, access_bridge)
```
Note that this is expected to be AND; use list(list()) for OR (like the original).

afterthought2 commented on code/modules/modular_computers/file_system/program.dm:
```diff
@@ -83,32 +83,41 @@
+
+	for(var/access in I.access)
+		if(access in accesses_to_check)
+			return TRUE
```
Use the existing helper (`has_access`); this will correctly handle AND/OR usage in a way consistent with other access stuff.

afterthought2 reviewed (commented):
Note to self: the report access stuff should really be converted to the main system.
